Observer/ConfigReview.php

```diff
@@ -117,4 +118,20 @@ private function _reviewNexusAddresses()
             // @codingStandardsIgnoreEnd
         }
     }
+
+    /**
+     * @return void
+     * @SuppressWarnings(Generic.Files.LineLength.TooLong)
+     */
+    private function _reviewAddressValidation()
+    {
+        $enabled = $this->scopeConfig->getValue(TaxjarConfig::TAXJAR_ADDRESS_VALIDATION);
+        $prevEnabled = $this->cache->load('taxjar_salestax_config_address_validation');
+
+        if (isset($prevEnabled) && $prevEnabled != $enabled) {
+            // @codingStandardsIgnoreStart
+            $this->messageManager->addErrorMessage(__('Please redeploy production mode in the Magento CLI ($ bin/magento deploy:mode:set production) to ensure address validation works correctly.'));
```

Comment:
@ryanreeves-taxjar Just tested this locally, I would go with a warning message via `addWarningMessage` instead of an error message here.